_posts/2025-04-25-from-recipes-to-vectors-building-a-hybrid-search-app-with-opensearch.md

```diff
@@ -0,0 +1,299 @@
+---
+layout: post
+title:  "Recipes to vectors: Building a hybrid search app with OpenSearch"
+authors:
+   - lizozom
+date: 2025-04-25
+categories:
+  - technical-posts
+meta_keywords: hybrid search, vector search, semantic search, OpenSearch
+meta_description: Learn how to build a hybrid search application from scratch using a publiclly available recipe data set from Kaggle. This guide explores data preparation, vectorization, and the use of OpenSearch for efficient search and retrieval.
+---
```

Comment:
Updated meta: 
_meta_keywords:_ hybrid search, vector search, open source vector database,  semantic search tutorial, semantic and keyword search, embedding models, OpenSearch k-NN plugin, search relevance, HNSW indexing

_meta_description:_ Learn how to build a hybrid search application using OpenSearch as a vector database. Bookmark this step-by-step guide to implementing semantic and keyword search with a recipe dataset.